### PR TITLE
docs: update migration guide for major bump release

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Update migration guide for major bump release
- angular
- react
- vue

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by reesscot** [comment]
https://next-release-migration-docs.d22cn4asamk913.amplifyapp.com/

File: docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx [hunk 1]
```diff
@@ -1,6 +1,39 @@
...
+
+### Update and usage
+
+`@aws-amplify/ui-angular@4.x` introduces the following breaking changes:
+
+#### 1. `@aws-amplify/ui-angular@4.x` bumps minimum Angular version to 14
+
+If your application is on Angular 13 or older, please migrate to Angular 14+ as per [official guide](https://update.angular.io).
+
+*Note*: Going forward, `@aws-amplify/ui-angular` will have the same version support as Angular's official LTS [support window](https://angular.io/guide/releases#actively-supported-versions).
```
Comment (thaddmt) [inline docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:25]: Nit: maybe to make it abundantly clear he we can say something to the effect of `starting from version x` isntead of `going foward`?

Comment (sreeramsama) [reply docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:25]: I think we can remove this note as we have aligned to support Angular's official LTS with a note in previous version upgrade.

Comment (reesscot) [reply docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:25]: Or we could just say that we support the current LTS versions in our Angular installation page?

Comment (wlee221) [reply docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx:25]: I agree with @reesscot's suggestion.

Comment (sreeramsama) [reply docs/src/pages/[platform]/getting-started/migration/migration.angular.mdx]: Updated in https://github.com/aws-amplify/amplify-ui/pull/4007/commits/cda137e4bd7d2c6cd8e8c0e360b59af3e354b205

File: docs/src/pages/[platform]/getting-started/installation/dependencies.angular.mdx [hunk 2]
```diff
@@ -4,3 +4,5 @@ import { InstallScripts } from '@/components/InstallScripts.tsx';
 Install `@aws-amplify/ui-angular` with [npm](https://www.npmjs.com/) or [yarn](https://yarnpkg.com/):
 
 <InstallScripts framework="angular" />
+
+*Note*: `@aws-amplify/ui-angular` will have the same version support as Angular's official LTS [support window](https://angular.io/guide/releases#actively-supported-versions).
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/installation/dependencies.angular.mdx:8]:
```suggestion
*Note*: `@aws-amplify/ui-angular` support aligns with Angular's official LTS [support window](https://angular.io/guide/releases#actively-supported-versions).
```

**Review by reesscot — DISMISSED** [review]

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 3]
```diff
@@ -1,6 +1,66 @@
...
+</Tabs>
+
+#### 3. `@aws-amplify/ui-react@5.x` remove the `to` prop on `Link` component and instead have it extended from the underlying rendered third-party `Link` if it contains a `to` prop.
+
+For example, using React Router Link with Amplify Link.
+
+```
+import { Link as ReactRouterLink } from 'react-router-dom';
+
+<Link as={ReactRouterLink} to="/" >Home</Link>
```
Comment (calebpollman) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:59]: IMO this is a bit difficult to parse, suggest the following:
```suggestion
#### 3. `@aws-amplify/ui-react@5.x` removes the `to` prop from the `Link` component.

If using a `Link` component with the `to` prop, remove `to` and instead extend from the underlying rendered third-party `Link`. For example, using React Router `Link` with the Amplify UI `Link` component:

```tsx
import { Link } from '@aws-amplify/ui-react';
import { Link as ReactRouterLink } from 'react-router-dom';

<Link as={ReactRouterLink} to="/">Home</Link>
```

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 4]
```diff
@@ -1,6 +1,66 @@
...
+<Tabs>
+<TabItem title="npm">
+<TerminalCommand command="npm install @aws-amplify/ui-react-geo@1.x" />
+</TabItem>
+<TabItem title="yarn">
+<TerminalCommand command="yarn add @aws-amplify/ui-react-geo@1.x" />
+</TabItem>
+</Tabs>
+
+Styles supporting Amplify Geo UI components are moved to `@aws-amplify/ui-react-geo` library and are not available through `@aws-amplify/ui-react` library anymore.
```
Comment (calebpollman) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:33]: Can we add an example?

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 5]
```diff
@@ -1,6 +1,66 @@
...
+When using components like `MapView` or `LocationSearch` from `@aws-amplify/ui-react-geo` library, import the styles from the same library as well.
+
+```
+import '@aws-amplify/ui-react-geo/styles.css';
+```
+
+#### 2. `@aws-amplify/ui-react@5.x` drops Amplify Notifications components. They are moved to a separate new package `@aws-amplify/ui-react-notifications`.
+Install the 1.x version of the `@aws-amplify/ui-react-notifications` library and update your existing imports accordingly.
```
Comment (calebpollman) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:41]: Can we add an example?

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx (hunk 4 above)
Comment (sreeramsama) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:33]: Added in https://github.com/aws-amplify/amplify-ui/pull/4007/commits/48000eb5eab6f7fb07df313fd6c7bf9d82423967

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx (hunk 5 above)
Comment (sreeramsama) [reply docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:58]: Added in https://github.com/aws-amplify/amplify-ui/pull/4007/commits/48000eb5eab6f7fb07df313fd6c7bf9d82423967

**Review by calebpollman — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**